docker_images/mbed-os-env/Dockerfile

```diff
@@ -0,0 +1,103 @@
+# ------------------------------------------------------------------------------
+# Pull base image
+FROM ubuntu:20.04
+
+# ------------------------------------------------------------------------------
+# Arguments
+ARG WORKDIR=/root
+
+# ------------------------------------------------------------------------------
+# Install tools via apt
+ENV DEBIAN_FRONTEND=noninteractive
+RUN set -x \
+    && apt -y update \
+    && apt -y install \
+    git \
+    wget \
+    python3 \
+    python3-dev \
+    python3-setuptools \
+    python3-usb \
+    python3-pip \
+    software-properties-common \
+    build-essential \
+    astyle \
+    mercurial \
+    ninja-build \
+    libssl-dev \
+    cargo \
+    flex \
+    bison \
+    && apt clean && rm -rf /var/lib/apt/lists \
+    && update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 \
+    && : # last line
+
+RUN set -x \
+    && pip3 install -U cmake
+
+WORKDIR /tmp/
+
+RUN set -x \
+    && git clone --depth=1 --single-branch --branch Release_1_8_14 https://github.com/doxygen/doxygen \
+    && cd doxygen \
+    && mkdir build \
+    && cd build \
+    && cmake -G "Unix Makefiles" .. \
+    && make \
+    && make install
+
+# Set up mbed environment
+WORKDIR /tmp/
+COPY requirements.txt .
+RUN set -x \
+    && pip3 install -r requirements.txt  \
+    && : # last line
+
+# ------------------------------------------------------------------------------
+# Install Python modules (which are not included in requirements.txt)
+RUN set -x \
+    && pip3 install -U \
+    mbed-cli \
+    mbed-tools \
+    && : # last line
+
+# ------------------------------------------------------------------------------
+# Install arm-none-eabi-gcc
+WORKDIR /opt/mbed-os-toolchain
+RUN set -x \
+    && [ "$(uname -m)" = "aarch64" ] && \
+        TARBALL="gcc-arm-none-eabi-9-2019-q4-major-aarch64-linux.tar.bz2" || \
+        TARBALL="gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2" \
+    && wget -q https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/RC2.1/${TARBALL} \
+    && tar -xjf ${TARBALL} \
+    && rm ${TARBALL} \
+    && : # last line
+
+# ------------------------------------------------------------------------------
+# Configure mbed build system
+RUN set -x \
+    && mbed config -G GCC_ARM_PATH /opt/mbed-os-toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/ \
+    && mbed toolchain -G -s GCC_ARM \
+    && : # last line
+
+# ------------------------------------------------------------------------------
+# Configure environment variables
+ENV MBED_GCC_ARM_PATH=/opt/mbed-os-toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/
+ENV PATH="${PATH}:${MBED_GCC_ARM_PATH}"
```

Comment:
With `arm-none-eabi-gcc` in the path like this, Mbed CLI 1 or 2 will pick it up without having to do funky stuff like `mbed config -G GCC_ARM_PATH`

Reply:
ok. thanks, will remove those funky stuff :) .